CONTRIBUTING.md

```diff
@@ -88,11 +94,183 @@ We recommend the following commit structure in the following order:
 1. Refactoring (no functional change)
 1. Meaningful behavioural changes
 
-Follow [this guide](https://chris.beams.io/posts/git-commit/) to ensure you're writing good commit messages.
+Follow [Chris Beams' article on how to write a Git commit
+message](https://chris.beams.io/posts/git-commit/) to ensure you're writing
+good commit messages.
+
+## Contribution Agreement
+
+Mbed previously required a signed contributor agreement in order to accept
+contributions. Mbed no longer requires such an agreement. Mbed projects (unless
+otherwise noted in the specific project) now accept contributions under
+Apache-2.0 directly. For more details, refer to [the Mbed OS contributing
+guide](https://os.mbed.com/docs/mbed-os/latest/contributing/index.html)
+
+Thank you for contributing to `mbed-tools`.
+
+# Development and Testing
+
+For development and testing purposes, it is essential to use a virtual
+environment. tox is a commonly used development tool that helps manage virtual
+environments.
+
+## Setup Python and tox
+
+`mbed-tools` is compatible with Python 3.6 or later.
+
+If you are on a Linux distribution, or macOS, you will find that Python comes
+pre-installed on your system. **Do not use the pre-installed versions of Python
+for development.**
+
+Below are links to guides for correctly setting up a development ready version of Python 3 on common platforms:
+
+* [MacOS](https://docs.python-guide.org/starting/install3/osx/#doing-it-right)
+* [Ubuntu](https://docs.python-guide.org/starting/install3/linux/)
+* [Windows](https://docs.python-guide.org/starting/install3/win/)
+
+On Windows we have also found the [Python Launcher for Windows](https://docs.python.org/3/using/windows.html) to be a reasonable alternative.
+
+After you have set up your Python 3 installation, install tox (the `--user` flag is important **never use `sudo` to install Python packages!**)
+
+```bash
+python3 -m pip install --user tox
```

Comment:
I'm wondering if we should tell macOS and Ubuntu users to `brew/apt install tox`. It's easier then faffing around installing Python then `pip install`ing `tox`, which is only needed on Windows. `brew` will kindly install `tox` in a venv for you and link the entry point to a directory in `PATH`. `apt` installs `tox` to the system, but the dependencies are other `apt` packages verified to work with Ubuntu's system Python installation, so it won't destroy your world.

Reply:
on macOS, most people using brew already have python installed (@3.8 AND @3.9) -- if `brew install tox` makes it easier and cleaner, I think it's best!